File: c sharp/Ej1/Estado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ej1
{
    public class Estado
    {
        public Estado()
        {
            arqueologosIzq = 0;
            canibalesIzq = 0;
            arqueologosDer = 0;
            canibalesDer = 0;
            linternaDer = false;
        }

        public int arqueologosIzq { get; set; }
        public int canibalesIzq { get; set; }
        public int arqueologosDer { get; set; }
        public int canibalesDer { get; set; }
        public bool linternaDer { get; set; }
    }
}

File: Estado.cpp
#include "Estado.h"

Estado::Estado(){
  this->arqueologosIzq = 0;
  this->canibalesIzq = 0;
  this->arqueologosDer = 0;
  this->canibalesDer = 0;
  this->linternaDer = true;
}

Estado::Estado(int arqueologosIzq, int canibalesIzq, int arqueologosDer, int canibalesDer, bool linternaDerecha){
  this->arqueologosIzq = arqueologosIzq;
  this->canibalesIzq = canibalesIzq;
  this->arqueologosDer = arqueologosDer;
  this->canibalesDer = canibalesDer;
  this->linternaDer = linternaDerecha;
}

void Estado::set_estado(int arqueologosIzq, int canibalesIzq, int arqueologosDer, int canibalesDer, bool linternaDerecha){
  this->arqueologosIzq = arqueologosIzq;
  this->canibalesIzq = canibalesIzq;
  this->arqueologosDer = arqueologosDer;
  this->canibalesDer = canibalesDer;
  this->linternaDer = linternaDerecha;
}